pkg/k8shandler/secret_test.go

```diff
@@ -0,0 +1,265 @@
+		cluster: &logging.ClusterLogging{},
+	}
+
+	// mimic ca.crt
+	caCrtStr := `-----BEGIN CERTIFICATE-----
+This is a ca cert.
+-----END CERTIFICATE-----`
+
+	// mimic ca.key
+	caKeyStr := `-----BEGIN PRIVATE KEY-----
+This is a private key.
+-----END PRIVATE KEY-----`
+
+	initSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
+			"testkey": []byte(caKeyStr),
+		})
+
+	// create
+	if err := clusterLoggingRequest.CreateOrUpdateSecret(initSecret); err != nil {
+		t.Errorf("CreateOrUpdateSecret failed 1 [%v]", err)
+	}
+
+	firstSecret, err := clusterLoggingRequest.GetSecret("test-secret")
+	if err != nil {
+		t.Errorf("GetSecret failed 1 [%v]", err)
+	}
+
+	caCrtStr = `-----BEGIN CERTIFICATE-----
+This is a new ca cert.
+-----END CERTIFICATE-----`
+
+	newSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
+			"testkey": []byte(caKeyStr),
+		})
+
+	// update
+	if err := clusterLoggingRequest.CreateOrUpdateSecret(newSecret); err != nil {
+		t.Errorf("CreateOrUpdateSecret failed 2 [%v]", err)
+	}
+
+	secondSecret, err := clusterLoggingRequest.GetSecret("test-secret")
+	if err != nil {
+		t.Errorf("GetSecret failed 2 [%v]", err)
+	}
+
+	if reflect.DeepEqual(firstSecret, secondSecret) {
+		t.Errorf("First secret [%v] == Second secret [%v]", firstSecret, secondSecret)
+	} else {
+		t.Logf("Initial secret [%p]\n%v", initSecret, initSecret)
+		t.Logf("Initial secret [%p]\n%v", newSecret, newSecret)
+		t.Logf("First secret [%p]\n%v", firstSecret, firstSecret)
+		t.Logf("Second secret [%p]\n%v", secondSecret, secondSecret)
+	}
+}
+
+func TestClusterLoggingRequest_CreateOrUpdateSecretNewKeySameCert(t *testing.T) {
+	clusterRoleBinding := &rbac.ClusterRoleBinding{}
+
+	clusterLoggingRequest := &ClusterLoggingRequest{
+		client:  fake.NewFakeClient(clusterRoleBinding),
+		cluster: &logging.ClusterLogging{},
+	}
+
+	// mimic ca.crt
+	caCrtStr := `-----BEGIN CERTIFICATE-----
+This is a ca cert.
+-----END CERTIFICATE-----`
+
+	// mimic ca.key
+	caKeyStr := `-----BEGIN PRIVATE KEY-----
+This is a private key.
+-----END PRIVATE KEY-----`
+
+	initSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
+			"testkey": []byte(caKeyStr),
+		})
+
+	// create
+	if err := clusterLoggingRequest.CreateOrUpdateSecret(initSecret); err != nil {
+		t.Errorf("CreateOrUpdateSecret failed 1 [%v]", err)
+	}
+
+	firstSecret, err := clusterLoggingRequest.GetSecret("test-secret")
+	if err != nil {
+		t.Errorf("GetSecret failed 1 [%v]", err)
+	}
+
+	caKeyStr = `-----BEGIN PRIVATE KEY-----
+This is a new private key.
+-----END PRIVATE KEY-----`
+
+	newSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
+			"testkey": []byte(caKeyStr),
+		})
```

Comment:
This looks like the SameKeyNewCert test

Reply:
Strange... why it did not happen on my laptop... :(

Reply:
> This looks like the SameKeyNewCert test

Where you could get the info?  This unit test log does not give me the details...
https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-logging-operator/238/pull-ci-openshift-cluster-logging-operator-master-unit/445
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-logging-operator/238/pull-ci-openshift-cluster-logging-operator-master-unit/445/build-log.txt

Reply:
This is the test `TestClusterLoggingRequest_CreateOrUpdateSecretNewKeySameCert`.  I would expect to have a new key e.g. `newca` instead of `testca` here, with the old value `This is a ca cert.` - that is, a new key, and the same cert value.